### PR TITLE
[core][autoscaler] Fix idle resource status accounting from raylet

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
There are 2 issues with idle resources accounting as of now: 
1. When we remove placement group resources, we delete them directly, which wasn't handled correctly in resources idleness accounting. 
2. When we block/unblock worker for ray.get or ray.wait, we temporarily change resources on the node, and we will make sure idleness is checked when such happens. 

Fixes this https://buildkite.com/ray-project/release-tests-pr/builds/48194#0189db35-8957-46c0-91cc-bdfd43dd5319

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by scv119 — APPROVED** [review]

File: src/ray/raylet/scheduling/local_resource_manager.cc [hunk 1]
```diff
@@ -53,6 +53,7 @@ void LocalResourceManager::AddLocalResourceInstances(
 void LocalResourceManager::DeleteLocalResource(scheduling::ResourceID resource_id) {
   local_resources_.available.Remove(resource_id);
   local_resources_.total.Remove(resource_id);
+  resources_last_idle_time_.erase(resource_id);
```
Comment (jjyao) [inline src/ray/raylet/scheduling/local_resource_manager.cc:56]: Let's do a sanity check to make sure we update resources_last_idle_time whenever OnResourceOrStateChanged is called.

Even better if we can update resources_last_idle_time inside OnResourceOrStateChanged so it's guaranteed that we don't miss anything.

Comment (rickyyx) [reply src/ray/raylet/scheduling/local_resource_manager.cc:56]: Ah, i see. i see this being a new function from your drain PR - we would want to pick this fix PR in 2.6.3, which might have some dependency with your drain PRs then?

Maybe we could address it and refactor later in master.

Comment (jjyao) [reply src/ray/raylet/scheduling/local_resource_manager.cc:56]: Yea, don't have to do it now.

File: src/ray/raylet/scheduling/local_resource_manager.cc [hunk 2]
```diff
@@ -239,37 +239,33 @@ void LocalResourceManager::FreeTaskResourceInstances(
...
   auto overflow =
       AddAvailableResourceInstances(resource_instances_fp,
                                     local_resources_.total.GetMutable(resource_id),
                                     local_resources_.available.GetMutable(resource_id),
-                                    &is_idle);
+                                    nullptr);
 
-  if (is_idle) {
-    SetResourceIdle(resource_id);
-  }
```
Comment (jjyao) [inline src/ray/raylet/scheduling/local_resource_manager.cc:259]: I don't think this fundamentally solve the issue: even if we avoid setting idle here, in the next resource update, we may still set it to idle.

File: src/ray/raylet/scheduling/local_resource_manager.cc [hunk 3]
```diff
@@ -238,37 +239,33 @@ void LocalResourceManager::FreeTaskResourceInstances(
...
 
-  bool is_idle = true;
   auto overflow =
       AddAvailableResourceInstances(resource_instances_fp,
                                     local_resources_.total.GetMutable(resource_id),
                                     local_resources_.available.GetMutable(resource_id),
-                                    &is_idle);
+                                    nullptr);
 
-  if (is_idle) {
```
Comment (rickyyx) [inline src/ray/raylet/scheduling/local_resource_manager.cc:257]: This is the major change for blocked worker cpu resource tracking: 
- we don't let temporary resources allocation change from blocked/unblocked worker to affect the resource idleness

Comment (rickyyx) [reply src/ray/raylet/scheduling/local_resource_manager.cc:257]: cc @jjyao

File: src/ray/raylet/scheduling/local_resource_manager.cc (hunk 2 above)
Comment (rickyyx) [reply src/ray/raylet/scheduling/local_resource_manager.cc]: ah yes - you are absolutely right.

**Review by rkooo567 — APPROVED** [review]

**Review by jjyao — APPROVED** [review]

**Pull request merged**